dotnet/src/Connectors/Connectors.Memory.InMemory/InMemoryCollection.cs

```diff
@@ -427,16 +427,17 @@ public override IAsyncEnumerable<TRecord> GetAsync(Expression<Func<TRecord, bool
             .AsQueryable()
             .Where(filter);
 
-        if (options.OrderBy.Values.Count > 0)
+        var orderBy = options.OrderBy?.Invoke(new()).Values;
+        if (orderBy is { Count: > 0 })
         {
-            var first = options.OrderBy.Values[0];
+            var first = orderBy[0];
```

Comment:
nit: should be able to use LINQ Aggregate() with OrderBy/OrderByDescending as the seed, and then composing ThenBy/ThenByDescending...